guidelines/backend-coding-standard/backend-coding-standard.md

```diff
@@ -0,0 +1,40 @@
+# Backend Coding Standard Draft
+
+## Coding style
+- function should be stateless, pure
+- function should not have side effect, i.e. interact with the environment outside the function, causing mutatation with other things that is not expected to change
+- input id instead of the object itself to the function, then query by id inside the function
+- function name should be describing the business action instead of implementation detail (except for private functions)
+  - e.g. prefer `consume_discount_code` to `decrease_discount_code_quota_by_one`
```

Comment:
I would say these rule should stand no matter it's public or private method.
To rephrase it, maybe these points can be split in 2 different points?

The first point should be more related to what we expected from the naming of module method(the name should be describing the business behaviour/action) and should be moved to the Modules section
The second point should be generic and suggest developer to always name their function/method with what(the behaviour) instead of how(the implementation).

And we should be able to provide some example for point 1 and point 2 (both do and don't).